manifests/charts/istio-control/istio-discovery/templates/mutatingwebhook.yaml

```diff
@@ -44,6 +44,9 @@ metadata:
     operator.istio.io/component: "Pilot"
     app: sidecar-injector
     release: {{ .Release.Name }}
+{{- if .Values.global.externalIstiod }}
+    externalIstiod: "true"
```

Comment:
what's the root cause, why need an additional label?

Reply:
The problem is that istiod doesn't have permission to acquire the leader lock on a non-istiodless remote cluster, so we need a way that istiod can check if it is an istiodless cluster before trying. Without checking the label, istiod will fail to acquire the lock when it doesn't have permission, which would be fine except that it generates error messages in the log. This PR is just needed to prevent the error messages from appearing in the log when istiod is connected to a non-istiodless remote cluster.

Reply:
I am confused since externalIstiod is always true now. why does istiod lack the permission?

https://github.com/istio/istio/blob/12f5ad29c4b83898077902f3e63826d2dd8d7d6f/manifests/charts/istiod-remote/templates/reader-clusterrole.yaml#L45-L55

Reply:
Ah, there are two different externalIstiod configuration parameters.

1. `values.pilot.env.EXTERNAL_ISTIOD` this is used to configure istiod running on an external or primary cluster. It controls whether or not istiod should attempt to manage other remote clusters. This is now true by default.
2. `values.global.externalIstiod` this is used to configure a remote cluster. True for istiodless remotes, and false by default.

It would have been clearer if the two weren't using the same name, but that was done a long time ago, so we're struck with it.

Reply:
Maybe the label can be the name(s) of primary cluster that should attempt to get the lock. The topology matters, and any EXTERNAL_ISTIOD cluster can't just acquire the lock for any externalIstiod cluster.

Reply:
Good idea. I added an annotation containing a regex of istiod clusters that should try.

Reply:
I don't feel comfortable matching cluster names with regex.. that feels like a recipe for headaches. Do we need multiple clusters matching?

Reply:
one cluster is enough but for HA I think we will want to allow multiple. If using regex for this is what concerns you, we could make it a list of cluster names instead. Would that be better?

Reply:
@howardjohn switched to one or more comma-separated cluster names

Reply:
Yeah was specifically regex. Thanks!